services/AttendanceService.ts

```diff
@@ -51,8 +51,36 @@ export default class AttendanceService {
       );
       await Repositories.user(txn).addPoints(user, pointsEarned);
 
-      const attendance = await attendanceRepository.attendEvent(user, event, attendedAsStaff);
-      return attendance.getPublicAttendance();
+      await attendanceRepository.attendEvent(user, event, attendedAsStaff);
+      return event.getPublicEvent();
+    });
+  }
+
+  public async attendEventRetroactively(userUuid: Uuid, eventUuid: Uuid): Promise<PublicEvent> {
+    return this.transactions.readWrite(async (txn) => {
+      const attendanceRepository = Repositories.attendance(txn);
+
+      const event = await Repositories.event(txn).findByUuid(eventUuid);
+      if (!event) throw new UserError('This event doesn\'t exist');
+      if (!event.hasStarted()) throw new UserError('This event hasn\'t started yet');
```

Comment:
Why this constraint? Could an event host potentially want to check someone in ahead of time?